clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2428,19 +2437,26 @@ static void emitUsed(CodeGenModule &CGM, StringRef Name,
   // Don't create llvm.used if there is no need.
   if (List.empty())
     return;
+  // Get target type when templated global variables are used,
+  // to emit them correctly in the target (default) address space and avoid
+  // emitting them in a private address space.
```

Comment:
How about something like
```suggestion
  // For SYCL emit pointers in the default address space which is a superset of other
  // address spaces, so casts from any other address spaces will be valid.
```